contracts/src/GetWethValueInPoolBatchRequest.sol

```diff
@@ -256,7 +256,7 @@ contract GetWethValueInPoolBatchRequest {
             ///@notice The first value is a Q96 representation of p_token0, the second is 128X fixed point representation of p_token1.
             uint256 priceSquaredShiftQ96 = token0IsReserve0
                 ? priceSquaredX96 / Q96
```

Comment:
Also is `Q96 > priceSquaredX96` here ever?

Reply:
i havent found the case when this happens to test it tho(( But im sure it can happen. What do you think we should do?

Reply:
```(priceSquaredX96 << 32) >> 64``` ?

Reply:
do `priceSquaredX96 << 128 / Q96`. Then below 

change 

```solidity
priceX128 = token0IsReserve0
                ? (uint256(priceSquaredShiftQ96) * 0xffffffffffffffffffffffffffffffff) / Q96
                : priceSquaredShiftQ96;
```

to 
```solidity
priceX128 = token0IsReserve0
                ? priceSquaredShiftQ96 / Q96
                : priceSquaredShiftQ96;
```

Reply:
We should really add some unit tests on this function after these changes. Maybe we could separate all peripheral functions into a deployable inherited contract that we can then easily unit test in foundry